DXMainClient/DXGUI/Multiplayer/GameInformationPanel.cs

```diff
@@ -194,17 +194,28 @@ public void SetInfo(GenericHostedGame game)
 
             if (mapLoader != null)
             {
-                mapTexture = mapLoader.GameModeMaps.Find(m => m.Map.UntranslatedName.Equals(game.Map, StringComparison.InvariantCultureIgnoreCase) && m.Map.IsPreviewTextureCached())?.Map?.LoadPreviewTexture();
+                Map map = mapLoader.GameModeMaps.Find(m => m.Map.UntranslatedName.Equals(game.Map, StringComparison.InvariantCultureIgnoreCase))?.Map;
+
+                if (map != null)
+                {
+                    if (map.IsPreviewTextureAvailableAsFile())
+                    {
+                        mapTexture = map.LoadPreviewTexture();
+                        disposeTextures = true;
+                    }
+                    else
+                    {
+                        mapTexture = AssetLoader.TextureFromImage(mapLoader.MapTextureCacheManager.GetMapTextureIfAvailable(map));
+                        disposeTextures = true;
```

Comment:
I am not sure you should dispose the cached textures though?

Reply:
Otherwise it would occupy VRAMs for those map previews. And a lot of users have limited VRAM. So only the bitmap is cached in RAM

Reply:
Oh right, it's not just a copy, nvm then.